docs/typescript.md

```diff
@@ -184,3 +225,30 @@ const Button = styled('button')`
 
 export default Button
 ```
+
+## preact-emotion
+
+Types are almost same with `react-emotion` package.
+
+## create-emotion
+
+Basically same with `emotion`, except that you can pass your own context and options.
+
+```jsx
+import createEmotion, { Emotion, EmotionOption } from 'create-emotion';
+
+const context = {};
+const options: EmotionOption = {
+  key: 'my-emotion',
+};
+const myEmotion: Emotion = createEmotion(context, options);
+
+const bodyStyle = myEmotion.css({
+  display: 'flex',
+  flowDirection: 'column-reverse',
+});
+```
+
+## create-emotion-styled
+
+The current typings for `create-emotion-styled` are only compatible with React, and will not work with Preact. For detail typing, see `react-element` section above.
```

Comment:
This isn't true anymore about not supporting Preact, right?

Reply:
FYI, Preact and React both modify `JSX.Element`, so we cannot import both typings in one file. That's why I did not provide typing for preact version of `create-emotion-styled`. (Its main typing file already import React typing)

Reply:
The only way I can image, to deal with the problem, is, providing index typing without import any (P)React related typings and making users import them.

However, this may confuse users, because it will not work properly when users do not import (P)React related typings.